packages/core/src/lib/social.ts

```diff
@@ -40,8 +40,14 @@ export const getUserInfoByAuthCode = async (
   authCode: string,
   redirectUri: string
 ): Promise<SocialUserInfo> => {
+  // TO-DO: rename and refactor connector methods
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  const data = JSON.parse(authCode);
```

Comment:
According to [this PR](https://github.com/logto-io/logto/pull/969/files#diff-006f79359ba081ed02adf6ba8a7c06514c1d5a4d5e1c17f7851a9af0ba1f0d42R70), this may cause a JSON syntax error if the `authCode` is not a JSON string.

Reply:
please go through the PR again. now the API will ALWAYS call with a JSON string